test/tuple.jl

```diff
@@ -533,6 +533,9 @@ end
 
     @test Base.setindex((1, 2, 4), 4, true) === (4, 2, 4)
     @test_throws BoundsError Base.setindex((1, 2), 2, false)
+
+    f() = Base._setindex(9, 1, (1:1, 2:2, 3:3)...)
```

Comment:
Is there any way to do this test without explicitly calling the internal `_setindex` function? What is the user facing difference this makes?

Reply:
Yes we can simply define `f() = Base.setindex((1:1, 2:2, 3:3), 9, 1)` instead ; I can make this change. The only reason why I called `Base._setindex` instead of `Base.setindex` is because the PR modifies the former method.